src/BoolUnification.js

```diff
@@ -117,6 +128,70 @@ function satisfiable(f) {
     }
 }
 
+/**
+ * Check whether an equation is of the form `Var(X) =? Equation`, where X not in free(Equation).
+ */
+function isSingleVarSpecialCase(maybeVar, maybeAssigned) {
+    return isVar(maybeVar) && !boolFreeVars(maybeAssigned).includes(maybeVar.name);
+}
+
+export function lowenheimUnify(f1, f2) {
+    if (f1 === undefined || !isBool(f1)) {
+        throw new Error(`Illegal argument 'x': ${f1}.`)
+    }
+    if (f2 === undefined || !isBool(f2)) {
+        throw new Error(`Illegal argument 'y': ${f2}.`)
+    }
+
+    // The boolean expression we want to show is 0.
+    let query = mkOr(mkAnd(f1, mkNot(f2)), mkAnd(mkNot(f1), f2))
+    // The free variables in the query.
+    let fvs = boolFreeVars(query)
+
+    let initUnifiers = initialUnifiers(query, fvs)
+    let mgus = initUnifiers.map(u => mguFromUnifier(u, query))
+
+    if (mgus.length === 0) {
+        return {status: "failure", reason: `Cannot unify: ${showBool(f1)} and ${showBool(f2)}`}
+    }
+
+    for (const [ind, mgu] of mgus.entries()) {
+        for (const [key, val] of Object.entries(mgu)) {
+            console.log(`  ${key} --> ${showBool(minBool(val, true))}`)
+        }
+    }
+
+    return {status: "success", subst: mgus[mgus.length -1]}
+}
+
+function initialUnifiers(query, fvs) {
+    let tt = truthTable(query, fvs)
+    let falseRows = tt.filter(t => isFalse(t[t.length - 1]))
+    let unifiers = falseRows.map(row => {
+        let subst = {};
+        for (const [index, elem] of row.slice(0, -1).entries()) {
+            subst[fvs[index]] = elem
+        }
+        return subst
+    })
+    return unifiers
+}
+
+function mguFromUnifier(unifier, term) {
```

Comment:
Add comment with a bit of intuition. At minimum explain what the function returns.